packages/@lwc/compiler/src/transformers/__tests__/transform-javascript.spec.ts

```diff
@@ -51,3 +55,92 @@ it('should object spread', async () => {
     expect(code).toContain('b: 1');
     expect(code).not.toContain('...a');
 });
+
+it('should transform unforgeables when Lightning Web Security is on', async () => {
+    const actual = `
+        export const test = window.location.href;
+    `;
+    TRANSFORMATION_OPTIONS.enableLightningWebSecurityTransforms = true;
+    const { code } = await transform(actual, 'foo.js', TRANSFORMATION_OPTIONS);
+
+    expect(code).toContain(
+        '(window === globalThis || window === document ? location : window.location).href'
+    );
+});
+
+it('should not transform unforgeables when Lightning Web Security is off', async () => {
+    const actual = `export const test = window.location.href;`;
+    TRANSFORMATION_OPTIONS.enableLightningWebSecurityTransforms = false;
+    const { code } = await transform(actual, 'foo.js', TRANSFORMATION_OPTIONS);
+    expect(stripWhitespace(code)).toMatch(stripWhitespace(actual));
+});
+
+it('should transform async await functions when Lightning Web Security is on', async () => {
+    const actual = `
+        export async function foo() {
+            await bar();
+        }
+    `;
+    TRANSFORMATION_OPTIONS.enableLightningWebSecurityTransforms = true;
+    const { code } = await transform(actual, 'foo.js', TRANSFORMATION_OPTIONS);
+
+    expect(stripWhitespace(code)).toContain(
+        stripWhitespace(`
+        function _asyncToGenerator(fn) {
+    `)
+    );
+
+    expect(stripWhitespace(code)).toContain(
+        stripWhitespace(`
+        _foo = _asyncToGenerator(function* () {
+            yield bar();
+        });
+    `)
+    );
+});
+
+it('should not transform async await functions when Lightning Web Security is off', async () => {
+    const actual = `
+        export async function foo() {
+            await bar();
+        }
+    `;
+    TRANSFORMATION_OPTIONS.enableLightningWebSecurityTransforms = false;
+    const { code } = await transform(actual, 'foo.js', TRANSFORMATION_OPTIONS);
+    expect(stripWhitespace(code)).toMatch(stripWhitespace(actual));
+});
+
+it('should transform async generator functions when Lightning Web Security is on', async () => {
+    const actual = `
+        async function* agf() {
+            await 1;
+            yield 2;
+        }
+    `;
+    TRANSFORMATION_OPTIONS.enableLightningWebSecurityTransforms = true;
+    const { code } = await transform(actual, 'foo.js', TRANSFORMATION_OPTIONS);
+
+    expect(stripWhitespace(code)).toContain(
+        stripWhitespace(`
+        function _agf() {
+            _agf = _wrapAsyncGenerator(function* () {
```

Comment:
we can simplify the sniff here to just `_agf = _wrapAsyncGenerator`.

Reply:
Because this is babel helper code we want to sniff as little as possible in case they change up their names